tests/pf/test_forward_Mag_Linear.py

```diff
@@ -1,77 +1,162 @@
-    nx = 5
-    ny = 5
+from SimPEG import maps, utils
+from SimPEG.potential_fields import magnetics as mag
 
-    H0 = (50000.0, 60.0, 250.0)
-    b0 = mag.analytics.IDTtoxyz(-H0[1], H0[2], H0[0])
-    chi1 = 0.01
-    chi2 = 0.02
 
+@pytest.fixture
+def mag_mesh() -> discretize.TensorMesh:
+    """
+    :return: TensorMesh for testing
+    """
     # Define a mesh
     cs = 0.2
     hxind = [(cs, 41)]
     hyind = [(cs, 41)]
     hzind = [(cs, 41)]
     mesh = discretize.TensorMesh([hxind, hyind, hzind], "CCC")
+    return mesh
 
-    # create a model of two blocks, 1 inside the other
-    block1 = np.array([[-1.5, 1.5], [-1.5, 1.5], [-1.5, 1.5]])
-    block2 = np.array([[-0.7, 0.7], [-0.7, 0.7], [-0.7, 0.7]])
-
-    def get_block_inds(grid, block):
-        return np.where(
-            (grid[:, 0] > block[0, 0])
-            & (grid[:, 0] < block[0, 1])
-            & (grid[:, 1] > block[1, 0])
-            & (grid[:, 1] < block[1, 1])
-            & (grid[:, 2] > block[2, 0])
-            & (grid[:, 2] < block[2, 1])
-        )
-
-    block1_inds = get_block_inds(mesh.cell_centers, block1)
-    block2_inds = get_block_inds(mesh.cell_centers, block2)
 
-    model = np.zeros(mesh.n_cells)
-    model[block1_inds] = chi1
-    model[block2_inds] = chi2
+@pytest.fixture
+def two_blocks() -> tuple[np.ndarray, np.ndarray]:
+    """
+    dimensions of two blocks
 
-    active_cells = model != 0.0
-    model_reduced = model[active_cells]
+    :return: tuple of (3, 2) arrays of (xmin, xmax), (ymin, ymax), (zmin, zmax) dimensions of each block
+    """
+    block1 = np.array([[-1.5, 1.5], [-1.5, 1.5], [-1.5, 1.5]])
+    block2 = np.array([[-0.7, 0.7], [-0.7, 0.7], [-0.7, 0.7]])
+    return block1, block2
 
-    # Create reduced identity map for Linear Pproblem
-    idenMap = maps.IdentityMap(nP=int(sum(active_cells)))
 
+@pytest.fixture
+def receiver_locations() -> np.ndarray:
+    """
+    grid of recievers for testing
+    :return: (n, 3) array of receiver locations
+    """
     # Create plane of observations
+    nx, ny = 5, 5
     xr = np.linspace(-20, 20, nx)
     yr = np.linspace(-20, 20, ny)
     X, Y = np.meshgrid(xr, yr)
     Z = np.ones_like(X) * 3.0
-    locXyz = np.c_[X.reshape(-1), Y.reshape(-1), Z.reshape(-1)]
-    components = ["bx", "by", "bz", "tmi"]
+    return np.c_[X.reshape(-1), Y.reshape(-1), Z.reshape(-1)]
+
+
+@pytest.fixture
+def inducing_field() -> tuple[tuple[float, float, float], tuple[float, float, float]]:
+    """
+    inducing field two ways
+    :return: (amplitude, inclination, declination), (b_x, b_y, b_z)
+    """
+    H0 = (50000.0, 60.0, 250.0)
+    b0 = mag.analytics.IDTtoxyz(-H0[1], H0[2], H0[0])
+    return H0, b0
+
+
+def get_block_inds(grid: np.ndarray, block: np.ndarray) -> np.ndarray:
+    """
+    get the indices for a block
+
+    :param grid: (n, 3) array of xyz locations
+    :param block: (3, 2) array of (xmin, xmax), (ymin, ymax), (zmin, zmax) dimensions of the block
+    :return boolean array of
```

Comment:
Docstrings in these utils functions and fixtures are much appreciated! In SimPEG we use [numpydoc style](https://numpydoc.readthedocs.io/en/latest/format.html). You might see some other styles in SimPEG's docstrings, but we are enforcing numpydocs in new PRs.

Would you mind changing them to numpydocs style? I think it can be automated with tools like [`pydocstring`](https://github.com/robodair/pydocstring)

Reply:
sure, no problem!